src/helpers/messaging.rs

```diff
@@ -78,14 +109,13 @@ impl<N: Network> Mesh<'_, '_, N> {
         record_id: RecordId,
         msg: T,
     ) -> Result<(), Error> {
-        let bytes = serde_json::to_vec(&msg)
-            .map_err(|e| Error::serialization_error(record_id, self.step, e))?
-            .into_boxed_slice();
+        let mut c = Cursor::new(Vec::with_capacity(T::BYTES as usize));
```

Comment:
Is the other allocation. Can we use smallvec here This

Reply:
If we want to cut an allocation we can do that later

Reply:
totally, that will be the end goal. did not want to change it here because `SendBuffer` will be affected as well as `Network` trait.